api/client/cp.go

```diff
@@ -191,10 +193,29 @@ func (cli *DockerCli) copyFromContainer(srcContainer, srcPath, dstPath string, c
 		_, srcBase := archive.SplitPathDirEntry(srcInfo.Path)
 		preArchive = archive.RebaseArchiveEntries(content, srcBase, srcInfo.RebaseName)
 	}
+
+	var reader io.ReadCloser
+	if cli.isTerminalOut {
+		size := stat.Size
+		if stat.Mode.IsDir() {
+			size = stat.DSize
+		}
+		sf := streamformatter.NewStreamFormatter()
+		progressOutput := sf.NewProgressOutput(cli.out, true)
+		reader = progress.NewProgressReader(preArchive, progressOutput, size, "", "Copying")
+		defer func() {
+			// Format blank line
+			cli.out.Write(sf.FormatStatus("", ""))
```

Comment:
My understanding is that `newLines` causes the progress output to emit a new line when the progress reader is finished. It looks very similar to what you're doing here.

https://github.com/docker/docker/blob/master/pkg/streamformatter/streamformatter.go#L126
https://github.com/docker/docker/blob/master/pkg/progress/progressreader.go#L40